cvpods/engine/hooks.py

```diff
@@ -129,6 +130,33 @@ def after_step(self):
             self._after_step(self.trainer)
 
 
+class OptimizationHook(HookBase):
```

Comment:
这个东西概念上不太对，你在after step的部分做了step？

Reply:
make optimizer a hook, such that sub_division and half precision, as well as other tricked applied to optimizer can be implemented in hook, rather than modifying the runner.py.

Reply:
建议说明一下必须要把optimizer放在register hook的第一个